### PR TITLE
Doc: Update/add/fix type annotations in ensemble+predict

### PR DESCRIPTION
**Pull request by dweindl**
Adding type annotations, fixing some typos.

### CODE REVIEW

File: pypesto/predict/amici_predictor.py [hunk 1]
```diff
@@ -191,9 +193,8 @@ def __call__(
 
         Returns
         -------
-        results:
-            PredictionResult object containing timepoints, outputs, and
-            output_sensitivities if requested
+        PredictionResult object containing timepoints, outputs, and
+        output_sensitivities if requested.
```
Comment (dilpath) [inline pypesto/predict/amici_predictor.py:197]:
```suggestion
        output sensitivities if requested.
```

**Review by dilpath — APPROVED** [review]
Thanks!

File: pypesto/predict/amici_predictor.py [hunk 2]
```diff
@@ -266,11 +267,17 @@ def __call__(
...
         mode: ModeType = MODE_FUN,
         include_llh_weights: bool = False,
         include_sigmay: bool = False,
-    ) -> Tuple[List, List, List]:
+    ) -> tuple[
+        list[np.array],
+        list[np.array],
+        list[np.array],
+        list[np.array],
+        list[np.array],
```
Comment (PaulJonasJost) [inline pypesto/predict/amici_predictor.py:279]: is there no nicer way to write this? 🤔

**Review by PaulJonasJost — APPROVED** [review]
Thanks 👍🏼

File: pypesto/predict/amici_predictor.py (hunk 2 above)
Comment (dweindl) [reply pypesto/predict/amici_predictor.py:279]: Not without loss of information.

**Pull request merged**